src/test/java/api/requests/scenarios/LoanDueDatesAfterRecallTests.java

```diff
@@ -0,0 +1,398 @@
+    assertThat("due date is the MGD (2 weeks)",
+        storedLoan.getString("dueDate"), is(expectedDueDate));
+  }
+
+  @Test
+  public void recallRequestWithMGDAndRDValuesChangesDueDateToMGDWithCLDDM()
+      throws InterruptedException,
+      ExecutionException,
+      TimeoutException,
+      MalformedURLException {
+    final IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final UUID checkOutServicePointId = UUID.fromString(CASE_FRI_SAT_MON_SERVICE_POINT_ID);
+    final IndividualResource steve = usersFixture.steve();
+    final IndividualResource jessica = usersFixture.jessica();
+
+    final LoanPolicyBuilder canCirculateRollingPolicy = new LoanPolicyBuilder()
+        .withName("Can Circulate Rolling With Recalls")
+        .withDescription("Can circulate item With Recalls")
+        .rolling(Period.weeks(3))
+        .unlimitedRenewals()
+        .renewFromSystemDate()
+        .withRecallsMinimumGuaranteedLoanPeriod(Period.days(8))
+        .withRecallsRecallReturnInterval(Period.weeks(1))
+        .withClosedLibraryDueDateManagement(DueDateManagement.MOVE_TO_THE_END_OF_THE_NEXT_OPEN_DAY.getValue());
+
+    final IndividualResource loanPolicy = loanPoliciesFixture.create(canCirculateRollingPolicy);
+
+    useLoanPolicyAsFallback(loanPolicy.getId(),
+        requestPoliciesFixture.noAllowedTypes().getId(),
+        noticePoliciesFixture.activeNotice().getId());
+
+    servicePointsFixture.create(new ServicePointBuilder(checkOutServicePointId, "CLDDM Desk", "clddm", "CLDDM Desk Test", null, null, TRUE, null));
+
+    // We use the loan date to calculate the MGD
+//    final DateTime loanDate = DateTime.now(DateTimeZone.UTC);
+    final DateTime loanDate =
+        new DateTime(2019, DateTimeConstants.JANUARY, 25, 10, 0, DateTimeZone.UTC);
+
+    // Update the clock so this will all work
+    ClockManager.getClockManager().setClock(Clock.fixed(Instant.ofEpochMilli(loanDate.getMillis()), ZoneOffset.UTC));
+
+    final IndividualResource loan = loansClient.create(new LoanBuilder()
+        .open()
+        .withItemId(smallAngryPlanet.getId())
+        .withUserId(steve.getId())
+        .withLoanDate(loanDate)
+        .withDueDate(loanDate.plusWeeks(3))
+        .withCheckoutServicePointId(checkOutServicePointId));
+
+    final String originalDueDate = loan.getJson().getString("dueDate");
+
+    requestsFixture.placeHoldShelfRequest(smallAngryPlanet, jessica,
+        DateTime.now(DateTimeZone.UTC), "Recall");
+
+    final JsonObject storedLoan = loansStorageClient.getById(loan.getId()).getJson();
+
+    assertThat("due date is not the original date",
+        storedLoan.getString("dueDate"), not(originalDueDate));
+
+    final String expectedDueDate =
+        CASE_FRI_SAT_MON_SERVICE_POINT_NEXT_DAY
+          .toDateTime(END_OF_A_DAY, DateTimeZone.UTC).toString(ISODateTimeFormat.dateTime());
+
+//    String expectedDueDate = loanDate.plusWeeks(2).toString(ISODateTimeFormat.dateTime());
```

Comment:
Please remove commented out piece of code

Reply:
Yeah, I noticed that when I was updating the test. I've removed the comment.